### PR TITLE
[12.0][ADD] New module hr_attendance_schedule

### PR DESCRIPTION
**Pull request by Jerther**
This is essencially an updated and improved version of #365 , ready for 12.0, since I concentrate my efforts on 12.0.

The base hr_attendance module tracks time with second precision. We needed to round attendances to a schedule and approve attendances outside a schedule, so we created this. We were surprised that such addons did not exist already, knowing many industries with attendance systems that work that way.

Also, the time shown in kiosk mode was that of the client computer, which is often offset by several minutes compared to the server's. This was no big deal with second precision, but with roundings we had to make sure the time shown was correct.

### CODE REVIEW

File: hr_attendance_schedule/README.rst [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+===========
+
+Bugs are tracked on `GitHub Issues
+<https://github.com/OCA/hr/issues>`_. In case of trouble, please
+check there if your issue has already been reported. If you spotted it first,
+help us smash it by providing detailed and welcomed feedback.
+
+Credits
+=======
+
```
Comment (nikul-serpentcs) [inline hr_attendance_schedule/README.rst:57]:
```suggestion
Authors
~~~~~~~

* Métal Sartigan

```

File: hr_attendance_schedule/README.rst [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+Contributors
+------------
+
+* Jérôme Thériault, Métal Sartigan <jtheriault@metalsartigan.com>
```
Comment (nikul-serpentcs) [inline hr_attendance_schedule/README.rst]:
```suggestion
* Jérôme Thériault <jtheriault@metalsartigan.com>
```

File: hr_attendance_schedule/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+{
+    'name': "Attendance Schedule",
+    'version': '12.0.1.0.0',
+    'author': "Métal Sartigan, Odoo Community Association (OCA)",
+    "license": "AGPL-3",
+    'website': "http://www.metalsartigan.com",
```
Comment (nikul-serpentcs) [inline hr_attendance_schedule/__manifest__.py]: URL should be `https://github.com/OCA/hr`

File: hr_attendance_schedule/demo/hr_attendance_schedule.xml [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+  </record>
+
+  <record id="hr_attendance_schedule_leave_1" model="resource.calendar.leaves">
+    <field name="name">Test leave</field>
+    <field name="calendar_id" ref="res_calendar_demo_schedule" />
+    <field name="date_from">2017-03-01 00:00:00</field>
+    <field name="date_to">2017-03-31 00:00:00</field>
+  </record>
+
+</odoo>
```
Comment (nikul-serpentcs) [inline hr_attendance_schedule/demo/hr_attendance_schedule.xml]: add new blank line

File: hr_attendance_schedule/views/res_calendar_attendance.xml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+      <xpath expr="//group/div" position="after">
+        <field name="grace_before_checkin" />
+        <field name="grace_after_checkin" />
+        <field name="grace_before_checkout" />
+        <field name="grace_after_checkout" />
+      </xpath>
+    </field>
+  </record>
+
+</odoo>
```
Comment (nikul-serpentcs) [inline hr_attendance_schedule/views/res_calendar_attendance.xml]: same other file

**Review by nikul-serpentcs — CHANGES_REQUESTED** [review]
Improve Code

File: hr_attendance_schedule/README.rst (hunk 1 above)
Comment (Jerther) [reply hr_attendance_schedule/README.rst:57]: I took a look at another README.rst and it seems to be a generated file which uses files in a readme folder. How does it work? There's nothing about it in the [contribution doc](https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst)

Comment (Jerther) [reply hr_attendance_schedule/README.rst:57]: Ok I found some info [here](https://github.com/OCA/maintainer-tools#readme-generator).

**Review by jarroyomorales — COMMENTED** [review]
This module is very interesting and useful, will you keep working on it? The only thing left to do to get travis green is to adjust files to the line limit marked by pep8 (79 characters per line)

**Comment by Jerther** [comment]
I'm really sorry, I really feel bad having to limit lines to 80 characters. 120 or even 100 would've been ok but I feel 80 is like going back to old school monochrome text only screens that could only display 80 characters per line (or 80 holes wide punch cards for that matter). That being said, I do respect the established codification standards and if that's all it takes to have the module accepted, I'll fix it.

**Comment by jarroyomorales** [comment]
@Jerther Yes :smile:  I also agree, but for now is what we've got. Thank you for understanding and fixing it!

**Comment by pedrobaeza** [comment]
From 13.0, you don't have to worry about line length. `black` auto-formatter decides for you the line length.

**Pull request closed without merging**